evm_loader/program/src/executor_state.rs

```diff
@@ -719,6 +722,7 @@ impl ExecutorSubstate {
 pub struct ExecutorState<'a, B: AccountStorage> {
     backend: &'a B,
     substate: Box<ExecutorSubstate>,
+    query_account_cache: query::AccountCache,
```

Comment:
`query_account_cache` is not saved in a storage account and will reset between iterations.
In order to automatically save it, cache should be part of `substate`.

Reply:
Moved the cache to the substate. And reworked Solidity part to inherit QueryAccount contract.